src/main/java/org/candlepin/subscriptions/tally/MetricUsageCollector.java

```diff
@@ -156,27 +159,38 @@ public MetricUsageCollector(
       }
     }
     accountRepository.save(account);
-    return accountCalcs;
+
+    return new CollectionResult(
+        new DateRange(effectiveStartDateTime, effectiveEndDateTime), accountCalcs, isRecalculating);
   }
 
   @Transactional
   public AccountUsageCalculation collectHour(Account account, OffsetDateTime startDateTime) {
     OffsetDateTime endDateTime = startDateTime.plusHours(1);
 
-    Stream<Event> eventStream =
+    Map<String, List<Event>> eventToHostMapping =
         eventController
             .fetchEventsInTimeRange(account.getAccountNumber(), startDateTime, endDateTime)
-            .filter(event -> event.getServiceType().equals(productProfile.getServiceType()));
+            .filter(event -> event.getServiceType().equals(productProfile.getServiceType()))
+            .collect(Collectors.groupingBy(Event::getInstanceId));
```

Comment:
It's not initially obvious what difference grouping the events by InstanceId has aside from avoiding the `account.getServiceInstances().get()` call. Please leave a note that we do it this way so that we can clear the measurements on first access. As an aside it occurs to me that if we have memory concerns with the number of events, we could instead keep a `Set<String>` of seen instances. Thoughts?

Reply:
Will do.

Regarding the memory concern, IMO since this should _rarely_ be used, I'm thinking that we leave it as is and revisit if/when we need to.

Reply:
:+1: